crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -178,6 +201,99 @@ fn create_diagnostic(
+        return ctx.diagnostic(only_used_in_recursion_diagnostic(property.span(), property_name));
+    }
+
+    let property_ident =
+        property.value.get_binding_identifier().expect("`bind_identifier` should be set");
+    let mut references = ctx
+        .semantic()
+        .symbol_references(property_ident.symbol_id.get().expect("symbol should be set"));
+
+    let has_spread_attribute = references.any(|x| used_with_spread_attribute(x.node_id(), ctx));
+
+    if has_spread_attribute {
+        // If the JSXElement has a spread attribute, we cannot apply a fix safely,
+        // as the same property name could be exist within the spread attribute.
+        return ctx.diagnostic(only_used_in_recursion_diagnostic(property.span(), property_name));
+    }
+
+    let Some(property_name) = property.key.static_name() else {
+        return;
+    };
+
+    ctx.diagnostic_with_dangerous_fix(
+        only_used_in_recursion_diagnostic(property.span, &property_name),
+        |fixer| {
+            let mut fix = fixer.new_fix_with_capacity(references.count() + 1);
+
+            let source = ctx.source_text();
+            let span_start = skip_to_next_char(source, property.span.start, &Direction::Backward)
+                .unwrap_or(property.span.start);
+            let span_end =
+                skip_to_next_char(ctx.source_text(), property.span.end, &Direction::Forward)
+                    .unwrap_or(property.span.end);
+
+            fix.push(Fix::delete(Span::new(span_start, span_end)));
+
+            let property_id = property
+                .value
+                .get_binding_identifier()
+                .expect("`binding identifier` should be set");
+
+            // search for references to the function and remove the property
+            for reference in ctx
+                .semantic()
+                .symbol_references(property_id.symbol_id.get().expect("symbol should be set"))
+            {
+                let mut ancestors = ctx.nodes().ancestors(reference.node_id());
+
+                let Some(attr) =
+                    ancestors.find_map(|node| match ctx.nodes().get_node(node).kind() {
+                        AstKind::JSXAttributeItem(attr) => Some(attr),
+                        _ => None,
+                    })
+                else {
+                    continue;
+                };
+
+                fix.push(Fix::delete(attr.span()));
+            }
+
+            fix
+        },
+    );
+}
+
+fn used_with_spread_attribute(node_id: NodeId, ctx: &LintContext) -> bool {
+    let opening_element = ctx
+        .nodes()
+        .ancestors(node_id)
+        .find_map(|node| match ctx.nodes().kind(node) {
+            AstKind::JSXOpeningElement(opening_element) => Some(opening_element),
+            _ => None,
+        })
+        .expect("should be safe");
+
+    opening_element
+        .attributes
+        .iter()
+        .any(|attr| matches!(attr, JSXAttributeItem::SpreadAttribute(_)))
```

Comment:
- we could avoid the `expect` call here by making the `.any()` the main part of the call
- we can use `iter_parents` to work with `Node`s directly

check my syntax, but something like this:

```suggestion
    ctx
        .nodes()
        .iter_parents(node_id)
        .any(|node| match node.kind() {
            AstKind::JSXOpeningElement(opening_element) => {
              opening_element
                .attributes
                .iter()
                .any(|attr| matches!(attr, JSXAttributeItem::SpreadAttribute(_)))
            },
            _ => false,
        })


    
```